### DISCUSSION TITLE
Data serialization Pydantic to Xarray

### DISCUSSION THREAD

**Discussion by DanielGuo1** [pydantic/pydantic#3630] [Question]
Hello,
is there a way to serialize Pydantic models into Xarray?

**Comment by NowanIlfideme** [comment]
Not built-in, no. There isn't a pre-defined way to do this, because you could be meaning any one of many things... 

1. Do you want to save the model in an `attrs` field? If so, you can just convert the model to JSON and parse it back, from a single field.
2. Do you want to save the model fields in `attrs` entirely? You just define converters (e.g. JSON for models and str() for simple types) to/from the `attrs` dict, which should be pretty simple.
3. Do you want to save values from pydantic into the xarray.DataArray values? I don't think you should do this, because you'd have to represent matrices as arrays of (arrays of...) values, which in Python is lists. If you have a super simple case where all your xarray values are scalars... you probably don't need xarray here. Data type conversion might also be a problem. You can try a go-between format, such as just using a Dict, and validating only fields that you actually use.
